swift/swift.py

```diff
@@ -52,102 +61,131 @@ def __init__(self, setup_env: dict, parent=None):
+        self.load(setupEnv["app"], setupEnv["bus"])
+        self.mainWindow.show()
 
-    def _init_bus(self, setup_bus: dict):
-        """Initializes global buses using set-up environment.
+    def load(self, appInfos: dict, busInfos: dict):
+        """Initializes swift system and loads the apps and buses.
         
         Args:
-            setup_bus: Set-up environment about bus.
+            appInfos: See "app" part of setupEnv in self.__init__().
+            busInfos: See "bus" part of setupEnv in self.__init__().
         """
-        for name, info in setup_bus.items():
-            # create a bus
-            if "timeout" in info:
-                timeout = info["timeout"]
-                bus = Bus(name, timeout)
-            else:
-                bus = Bus(name)
-            # set a slot of received signal to router
-            bus.received.connect(self._route_to_app)
-            bus.start()
-            # store the bus
-            self._buses[name] = bus
-            self._subscribers[name] = []
-
-    def _init_app(self, setup_app: dict):
-        """Initializes apps using set-up environment.
+        for name, info in busInfos.items():
+            self.createBus(name, info)
+        for name, info in appInfos.items():
+            self.createApp(name, info)
+
+    def createBus(self, name: str, info: dict):
+        """Creates a global bus using set-up environment.
         
         Args:
-            setup_app: Set-up environment about app.
+            name: A name of the bus.
+            info: A dictionary containing bus info. Each element is like below:
+              timeout: Desired timeout for blocking queue reading, in seconds.
+                If you want to set as default, set None.
         """
-        for name, info in setup_app.items():
-            # import the app module
-            path = info.get("path", ".")
-            mod_name, cls_name = info["module"], info["class"]
-            with _add_to_path(os.path.dirname(path)):
-                module = importlib.import_module(mod_name)
-            # create an app
-            cls = getattr(module, cls_name)
-            args = info.get("args", {})
-            app = cls(name, self, **args)
-            # set a slot of broadcast signal to router
-            app.broadcastRequested.connect(self._route_to_bus)
-            # add the app to the list of subscribers on each bus
-            for bus_name in info["bus"]:
-                self._subscribers[bus_name].append(app)
-            # store the app
-            self._apps[name] = app
-
-    def _show_frame(self, setup_app: dict):
-        """Shows frames of each app.
+        # create a bus
+        if "timeout" in info:
+            bus = Bus(name, info["timeout"])
+        else:
+            bus = Bus(name)
+        # set a slot of received signal to router
+        bus.received.connect(self._routeToApp)
+        bus.start()
+        # store the bus
+        self._buses[name] = bus

```

Comment:
I think it should be checked first that there is already a bus with name `name`.
If there were a bus already, it will be garbage collected in python, but not in Qt, and the queue consumer thread will not stop.
Maybe returning `False` when failed would be helpful.

Reply:
Or, the `name` existence checking should be done before calling this method... It seems a design choice.

Reply:
I see.. Actually, I'm going to talk about this.

In `Swift`, We only check key existence in a few cases.
Then, I suggest a solution; at the beginning part of `load()`, check if `appInfo` and `busInfo` have a correct structure.

Reply:
Umm.. How about checking `name` existence and if exists, just print a message?

Reply:
I think it should notify the caller the failure, with an exception or return value.
Printing a message is good (you can do this along with notifying the caller), but it only notifies the user, not the method caller.

Reply:
It's hard for me to understand your suggestion..😂
Did you mean that printing a message is enough?

Reply:
😄 sorry for unclarity, my opinion is:
Printing a message is okay, but in addition to that, raising an exception or returning the result is necessary.